rfcs/0024-wallet-internalization.md

```diff
@@ -0,0 +1,102 @@
+## Summary
+
+---
+
+Integrate an internationalization library into the Coda wallet for multi-language support.
+
+## Motivation
+
+---
+
+The Coda wallet currently has only English text, making it more difficult for non-English speaking developers and users to easily contribute and use.
+
+The wallet should support multiple languages to reach a wide-ranging audience.
+
+## Detailed design
+
+---
+
+React-intl is currently one of the most well-supported internationalization library for React. It is built upon the native **[Internationalization API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)** browser API, and neatly wraps them for performance and usability enhancements.
+
+Lib: [https://github.com/formatjs/react-intl](https://github.com/formatjs/react-intl)
+
+Bs-bindings: [https://github.com/reasonml-community/bs-react-intl](https://github.com/reasonml-community/bs-react-intl)
+
+General steps to implement react-intl
+
+1. Create a JSON of IDs mapped to their text. Each new language will have their own JSON file.
+
+Discussion: Am I converting _all_ texts in this grant? It's within capacity. Just not sure if the Coda team is comfortable with having an app-wide change immediately due to some potential internal conflicts.
```

Comment:
I would imagine we can work around this by having the initial PR target a few representative texts in order to iron out all the bugs/issues, then making a second PR that quickly swaps over the rest of text in order to minimize too many merge issues.

Reply:
That's a fair judgement. Let's do that. Do you have any screens in mind yet?

Reply:
I think the goal will be to eventually swap them all over, so I don't think it matters much, any text on the main view of the wallet should be fair game, maybe the settings page has more text so we can do that after (still as part of this RFC).

Could start with the main transaction listview component -- that covers the text in the headers as well as the date formatting logic I mentioned in my earlier comment.